waspc/cli/src/Wasp/Cli/Command/ShellCompletion.hs

```diff
@@ -0,0 +1,159 @@
+module Wasp.Cli.Command.ShellCompletion
+  ( completion,
+  )
+where
+
+import Control.Monad.IO.Class (liftIO)
+import System.Environment (getProgName)
+import Wasp.Cli.Command (Command)
+import Wasp.Cli.Command.Call
+  ( CompletionArgs (..),
+    Shell (Bash, Fish, Zsh),
+  )
+
+completion :: CompletionArgs -> Command ()
+completion PrintInstruction = printShellCompletionInstruction
+completion (PrintScript shell) = printShellCompletionScript shell
+
+type ShellCompletionScript = (String -> String -> String)
```

Comment:
Oops! My mistake. Revealing the names does help expose a bug. :sweat_smile:  I've found out that `progn` is the running program name and `prog` is the `progn` absolute filepath. 

If I understand this correctly, current code implementation completion script won't work in situation where the running program is not in `PATH` because the actual completer need to invoke the waspc exe for optparse-applicative own internal completion flags such as `--bash-completion-word`, etc. This won't work if it can't find exe location! :fearful: 

For example, from optparse-applicative completion README: ```$(foo --bash-completion-script `which foo`)``` is pretty much equals to `bashCompletionScript foo fooAbsolutePath`.


So perhaps what I should do to fix:

1. First and foremost check if we have existing utility in waspc/cli to get the absolute path of current running program, if not I'll implement it on my own with `findExecutable` and StrongPath.
2. Replace `prog` value with the actual absolute filepath.


For names, I might go with these for now:
```haskell
type AbsProgFile = FilePath
type ProgName = ProgName
type ShellCompletionScriptBuilder = (AbsProgFile -> ProgName -> String)
```

Reply:
Pushed the fix into https://github.com/wasp-lang/wasp/pull/1202/commits/34aa3beaebbb043e2deb148e6468fcf5d7c06245

Reply:
Ok, certainly an improvement, and nice think catching a bug! Ah yes, names reveal much :D.

I would maybe go with `AbsProgPath` or `AbsProfFilePath` instead, but otherwise sounds good. Also providing a comment above ShellCompletionScriptBuilder explaining the relationship between AbsProgFile and ProgName would be helpful -> because those are referring to the same program but it might not be obvious.